src/fibaro-api.ts

```diff
@@ -92,8 +92,13 @@ export class FibaroClient {
   }
 
   composeURL(service) {
-    if (this.url) {
+    // if url is filled and starts with https:// or http://
+    if (this.url.includes('http') ) {
```

Comment:
Use `startsWith` instead of `include`.
In general, try to perform a complete validation of the fields: check if url contains a valid url or a valid ip address.
Some for host.

Reply:
Check it out now